### ISSUE TITLE
Reset proxy ports, if no domain is set on first Dockerfile deploy

### ISSUE THREAD

**Issue by mimischi** [dokku/dokku#2903]
Dokku will clear all values in `DOKKU_PROXY_PORT_MAP` and set the variable to `http:9000:9000`, when deploying an app for the first time via Dockerfile (where `9000` is the exposed port, specified in the Dockerfile). Subsequent deploys are fine.
*Note*: global vhosts are disabled and no domain is specified for the app:

```user@host:~$ dokku domains:report test
=====> test domains information
       Domains app enabled: false                    
       Domains app vhosts:                           
       Domains global enabled: false                    
       Domains global vhosts
```

### `dokku report`
```
-----> uname: Linux poc3 4.4.0-93-generic #116-Ubuntu SMP Fri Aug 11 21:17:51 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
              total        used        free      shared  buff/cache   available
Mem:          64426        4763       55859         126        3803       58755
Swap:         51245           0       51245
-----> docker version: 
Client:
 Version:      17.07.0-ce
 API version:  1.31
 Go version:   go1.8.3
 Git commit:   8784753
 Built:        Tue Aug 29 17:42:53 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.07.0-ce
 API version:  1.31 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   8784753
 Built:        Tue Aug 29 17:41:43 2017
 OS/Arch:      linux/amd64
 Experimental: false
-----> docker daemon info: 
Containers: 52
 Running: 25
 Paused: 0
 Stopped: 27
Images: 157
Server Version: 17.07.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 3addd840653146c90a254301d6c3a663c7fd6429
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-93-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 62.92GiB
Name: poc3
ID: ZE6S:Z4H7:NDYC:PX2O:G6NT:5OYB:4STS:KI76:KMHJ:DKKX:TN5K:PYJF
Docker Root Dir: /var/lib/docker
Debug Mode (client): true
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
-----> sigil version: 0.4.0
-----> herokuish version: 
herokuish: 0.3.32
buildpacks:
  heroku-buildpack-multi     v1.0.0
  heroku-buildpack-ruby      v168
  heroku-buildpack-nodejs    v99
  heroku-buildpack-clojure   v77
  heroku-buildpack-python    v99
  heroku-buildpack-java      v53
  heroku-buildpack-gradle    v23
  heroku-buildpack-scala     v78
  heroku-buildpack-play      v26
  heroku-buildpack-php       v122
  heroku-buildpack-go        v72
  heroku-buildpack-erlang    fa17af9
  buildpack-nginx            v8
-----> dokku version: 0.10.4
-----> dokku plugins: 
plugn: 0.3.0
  00_dokku-standard    0.10.4 enabled    dokku core standard plugin
  20_events            0.10.4 enabled    dokku core events logging plugin
  apps                 0.10.4 enabled    dokku core apps plugin
  build-env            0.10.4 enabled    dokku core build-env plugin
  certs                0.10.4 enabled    dokku core certificate management plugin
  checks               0.10.4 enabled    dokku core checks plugin
  common               0.10.4 enabled    dokku core common plugin
  config               0.10.4 enabled    dokku core config plugin
  docker-options       0.10.4 enabled    dokku core docker-options plugin
  domains              0.10.4 enabled    dokku core domains plugin
  enter                0.10.4 enabled    dokku core enter plugin
  git                  0.10.4 enabled    dokku core git plugin
  letsencrypt          0.8.6 enabled    Automated installation of let's encrypt TLS certificates
  logs                 0.10.4 enabled    dokku core logs plugin
  memcached            1.0.0 enabled    dokku memcached service plugin
  named-containers     0.10.4 enabled    dokku core named containers plugin
  nginx-vhosts         0.10.4 enabled    dokku core nginx-vhosts plugin
  plugin               0.10.4 enabled    dokku core plugin plugin
  postgres             1.0.0 enabled    dokku postgres service plugin
  proxy                0.10.4 enabled    dokku core proxy plugin
  ps                   0.10.4 enabled    dokku core ps plugin
  redis                1.0.0 enabled    dokku redis service plugin
  repo                 0.10.4 enabled    dokku core repo plugin
  shell                0.10.4 enabled    dokku core shell plugin
  ssh-keys             0.10.4 enabled    dokku core ssh-keys plugin
  storage              0.10.4 enabled    dokku core storage plugin
  tags                 0.10.4 enabled    dokku core tags plugin
  tar                  0.10.4 enabled    dokku core tar plugin
```

#### Link to the exact repository being deployed (if possible/applicable)
[https://github.com/mimischi/dokku-sentry](https://github.com/mimischi/dokku-sentry)

#### Dokku trace and steps to reproduce
[https://gist.github.com/mimischi/4f3a9ad6c5a4e0e4209f72b448f2f4dd](https://gist.github.com/mimischi/4f3a9ad6c5a4e0e4209f72b448f2f4dd)

#### Actual Results
`DOKKU_PROXY_PORT_MAP` is set to `http:9000:9000`. All previously set values are purged.

#### Expected Results
`DOKKU_PROXY_PORT_MAP` is left untouched with the previous set value `http:80:9000`.

#### Additional info
Specifying a domain before the first deploy will lead to the expected behaviour.

**Comment by josegonzalez** [comment]
Okay this is because you don't have a global vhost. We basically disable domains but in the process, also remove proxy mounts, etc. If you set a global vhost, this will work as you'd expect.

We can't disable that removal because its necessary for ensuring that apps *without* custom ports - such as buildpack apps - get routed to properly.

The way to fix this would be to always respond with the "default" detected mapping - `http:5000:9000` for buildpacks, whatever for dockerfile - unless something is set...

**Comment by josegonzalez** [comment]
Note: this is a duplicate of #4607 - that has more data - and will be completed in #5548

**Comment by josegonzalez** [comment]
Reopening as this hasn't yet been fixed.

**Comment by josegonzalez** [comment]
This will be implemented in 0.31.0 with the introduction of https://github.com/dokku/dokku/pull/6029, which will move these ports to a "detected" variable and allow users to override them as desired.
